scripts/gulp/frontend-shared.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const cp = require('child_process');
+const fs = require('fs');
+const gulp = require('gulp');
+const babel = require('gulp-babel');
+const sourcemaps = require('gulp-sourcemaps');
+
+const buildFrontendSharedJs = () => {
+  // There does not appear to be a simple way of forcing gulp-babel to use a config
+  // file. Load it up as JSON and pass it in manually.
+  const babelConfig = JSON.parse(
+    fs.readFileSync('./frontend-shared/babel.config.json')
+  );
+
+  return (
+    gulp
+      .src('frontend-shared/src/*')
+      // Transpile the js source files and write the output in the frontend-shared/lib dir.
+      // Additionally, add the sourcemaps into the same dir.
+      .pipe(sourcemaps.init())
+      .pipe(babel(babelConfig))
+      .pipe(sourcemaps.write('.'))
+      .pipe(gulp.dest('frontend-shared/lib'))
+  );
+};
+
+const linkFrontendShared = done => {
```

Comment:
This will try and re-link every time its run, basciallyonce each time you call `make dev`. It's not a huge cost, but it's a little clunky.  It may just be a good enough hack for the time being. the main thing I wanted was something that we don't have to think about or manage.